### PR TITLE
[Merged by Bors] - feat: add fluvio hub util

### PR DESCRIPTION
**Pull request by digikata**

### CODE REVIEW

File: crates/fluvio-hub-util/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+pub mod keymgmt;
+pub use package::*;
+pub use packagemeta::*;
+pub use errors::HubUtilError;
+pub use utils::*;
+
+pub const HUB_PACKAGE_VERSION: &str = "0.1";
+pub const HUB_PACKAGE_META: &str = "package-meta.yaml";
+
+pub const HUB_BUCKET: &str = "devel-hub";
```
Comment (sehz) [inline crates/fluvio-hub-util/src/lib.rs:26]: remove BUCKET REF

**Review by sehz — CHANGES_REQUESTED** [review]
first sets of comments

File: crates/fluvio-hub-util/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+features = ["h1-client-rustls", "middleware-logger", "encoding"]
+default-features = false
+
+[dependencies.http-client]
+version = "6.5.3"
+features = ["h1_client", "rustls"]
+default-features = false
+
+[dev-dependencies]
+env_logger = "*"
```
[the remark this answers is not in the document]
Comment (digikata) [reply crates/fluvio-hub-util/Cargo.toml]: nope, removed

File: crates/fluvio-hub-util/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+[package]
+name = "fluvio-hub-util"
```
[the remark this answers is not in the document]
Comment (digikata) [reply crates/fluvio-hub-util/Cargo.toml:2]: fluvio-hub-util handles smartmodule packages, but doesn't have to be only smartmodules. Maybe we should keep the name more general for now?

File: crates/fluvio-hub-util/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+version = "6.5.3"
+features = ["h1_client", "rustls"]
+default-features = false
+
+[dev-dependencies]
+env_logger = "*"
+test-log = { version = "0.2", features = ["trace"] }
+
+[dev-dependencies.tracing-subscriber]
+version = "0.3.15"
```
[the remark this answers is not in the document]
Comment (digikata) [reply crates/fluvio-hub-util/Cargo.toml]: I just removed all the dev dependencies as they weren't necessary.

File: crates/fluvio-hub-util/Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sehz) [reply crates/fluvio-hub-util/Cargo.toml:2]: k

**Comment by digikata** [comment]
Squashed and ready for next review & merge

File: crates/fluvio-hub-util/src/package.rs [hunk 5]
```diff
@@ -0,0 +1,443 @@
+//! # package
+
+//! phase1, unpackage file, verify signature
```
Comment (sehz) [inline crates/fluvio-hub-util/src/package.rs:3]: remove these comments

File: crates/fluvio-hub-util/src/utils.rs [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+            // phase 1 hub-dev
+            (
+                "example@0.0.1",
+                "https://hub.infinyon.cloud/hub/v0/pkg/pub/example/0.0.1",
+            ),
+            (
+                "infinyon/example@0.0.1",
+                "https://hub.infinyon.cloud/hub/v0/pkg/pub/infinyon/example/0.0.1",
+            ),
+            // phase 2 hub-dev
```
Comment (sehz) [inline crates/fluvio-hub-util/src/utils.rs:202]: is this comments needed?

File: crates/fluvio-hub-util/src/utils.rs [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+
+        // todo test let recs_bad
+    }
+
+    #[test]
+    fn cli_pkgname_to_filename_t() {
+        let recs_good = vec![
+            ("example@0.0.1", "example-0.0.1.ipkg"),
+            ("infinyon/example@0.0.1", "infinyon-example-0.0.1.ipkg"),
+            // support later, needs hub-pkg-server support
```
Comment (sehz) [inline crates/fluvio-hub-util/src/utils.rs:226]: remove these comments

**Review by sehz — CHANGES_REQUESTED** [review]
A couple of nits.   Need to clean up logging and other usages but we can at post-commit

**Comment by digikata** [comment]
one more cleanup round - resolved all comments and a few more cleanup areas. Resquashed

**Comment by sehz** [comment]
CI failing

**Comment by digikata** [comment]
minor test fix, rebased

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**